tests/ui/empty_enum_variants_with_brackets.rs

```diff
@@ -16,6 +17,20 @@ enum TestEnum {
     AnotherEnum,                       // No error
 }
 
+mod issue12551 {
+    enum EvenOdd {
```

Comment:
It would be good to have a test where a variant is used as a function before its definition, to make sure it doesn't overwrite a `Used` entry with `Unused`